quickwit/quickwit-ui/src/components/QueryEditor/AggregationEditor.tsx

```diff
@@ -0,0 +1,316 @@
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.field = value;
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  const handleTermCountChange = (pos: number, event: React.ChangeEvent<HTMLInputElement|HTMLTextAreaElement>) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.size = Number(value);
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  function isHistogram(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {histogram: HistogramAgg} {
+    if (!agg) return false;
+    return "histogram" in agg;
+  }
+
+  function isTerm(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {term: TermAgg} {
+    if (!agg) return false;
+    return "term" in agg;
+  }
+
+  const getAggregationKind = (agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined) => {
+    if (isHistogram(agg)) {
+      return "histogram";
+    }
+    if (isTerm(agg)) {
+      return "term";
+    }
+    return "new";
+  };
+
+  const makeOptions = (pos: number, agg: ({term: TermAgg} | {histogram: HistogramAgg})[]) => {
+    const options = [];
+    if (pos >= agg.length) {
+      options.push((
+          <MenuItem value="new" key="new">Add aggregation</MenuItem>
+      ))
+    }
+    let addHistogram = true;
+    let addTerm = true;
+    for(let i = 0; i < agg.length; i++) {
+      if (i == pos) continue;
+      if (getAggregationKind(agg[i]) === "histogram") addHistogram = false;
+      if (getAggregationKind(agg[i]) === "term") addTerm = false;
+    }
+    if (addHistogram) {
+      options.push((<MenuItem value="histogram" key="histogram">Histogram aggregation</MenuItem>))
+    }
+    if (addTerm) {
+      options.push((<MenuItem value="term" key="term">Term aggregation</MenuItem>));
+    }
+    if (agg.length > 1) {
+      options.push((
+          <MenuItem value="rm" key="rm">Remove aggregation</MenuItem>
+      ))
+    }
+    return options;
+  }
+
+  // do the initial filling of parameters
+  const aggregationConfig = props.searchRequest.aggregationConfig;
+  if (aggregationConfig.histogram === null && aggregationConfig.term === null) {
+    const initialAggregation = Object.assign({}, ...aggregations);
+    const initialSearchRequest = {...props.searchRequest, aggregationConfig: initialAggregation};
+    props.onSearchRequestUpdate(initialSearchRequest);
+  }
+
+  const drawAdditional = (pos: number, aggs: ({term: TermAgg} | {histogram: HistogramAgg})[]) => {
+    const agg = aggs[pos]
+    if (isHistogram(agg)) {
```

Comment:
This may very well crash the tab of the unsuspecting user, trying to render thousands of buckets.
Search params being serialized in the urls make it even worse (refresh won't help).

I'd favor computing a safe-to-render resolution based on the selected time range.

Reply:
so i should remove that Select? What should i do when no time range is selected? Send 2 requests for the lowest/highest timestamp matching the query?

Reply:
I think the 2nd option would be more friendly. Tutorials are made with "historical" data, where a default range would likely match nothing. 3 is a worst version of 2 imo. 4 sounds bad to use given the simple search allows not providing bounds

Reply:
I started to implement the 2nd option on a private branch but it seems to add a lot of complexity, potention fetch errors, and lots of lignes of code to the merge requests. So went with the "removing the option here is probably the best course" solution. I did put 1day interval by default to reduce to probability of errors due to the number of buckets selected.